frontend/sw360-portlet/src/main/resources/META-INF/resources/js/modules/ajax-treetable.js

```diff
@@ -17,23 +17,48 @@ define('modules/ajax-treetable', [ 'jquery', 'utils/cssloader', /* jquery-plugin
     return {
         setup: function(tableId, ajaxUrl, dataCallback, renderCallback) {
             var table = $("#" + tableId);
+            var nodeCount = 0;
             table.treetable({
                 expandable: true,
                 onNodeExpand: function () {
+                    var resultCount = 0;
+                    var areAllNodesExpanded = false;
                     var node = this,
                         data = dataCallback(table, node);
 
-                    if (node.children.length === 0 && node.row.data("children-loaded") !== true) {
+                    if(nodeCount == Object.keys(node.tree).length && $('#LinkedProjectsInfo #expandAll').is(':visible')
+                       && !$('#LinkedProjectsInfo tr').hasClass('branch collapsed')){
+                       $('#LinkedProjectsInfo #expandAllWarning').show();
+                    }
+
+                    if (node.children.length === 0 && node.row.data("children-loaded") !== true ) {
                         jQuery.ajax({
                             type: 'POST',
                             url: ajaxUrl,
                             cache: false,
                             data: data
                         }).done(function (result) {
                             node.row.data("children-loaded", true);
+                            
+                            if(/[a-z]/i.test(result)){
+                               resultCount++;
+                            }
+                            nodeCount++;
+
+                            if(nodeCount == Object.keys(node.tree).length){
```

Comment:
please use correct indentation / formatting.
`if (condition) {` -> at line `52`, `43` and `29`.